### PR TITLE
pscanrulesBeta: Update to take advantage of 2.10 core

### PR DESCRIPTION
**Pull request by kingthorin**
- Update scan rules to leverage new Custom Pages functionality.
- Update scan rule unit tests to account for the changes.
- Update build file to target ZAP 2.10 and leverage the core snapshot library.
- Update CHANGELOG with a change note about the updates.

Part of zaproxy/zaproxy#9

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: addOns/pscanrulesBeta/CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,14 @@ All notable changes to this add-on will be documented in this file.
...
 
 ## Unreleased
-
+### Changed
+- Now targeting ZAP 2.10.
+- The following scan rules now support Custom Page definitions:
+  - Insecure Form Load
+  - Insecure Form Post
+  - User Controlled Charset
+  - User COntrolled HTML Attribute
```
Comment (thc202) [inline addOns/pscanrulesBeta/CHANGELOG.md:13]: There's a typo in controlled (same for following line).

Comment (kingthorin) [reply addOns/pscanrulesBeta/CHANGELOG.md:13]: Good catch, fix coming.

**Comment by kingthorin** [comment]
Typo corrected.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thanks!

**Review by psiinon — APPROVED** [review]

**Pull request merged**